client/src/app/site/users/services/user-pdf.service.ts

```diff
@@ -0,0 +1,294 @@
+            },
+            {
+                text: this.translate.instant('Initial password') + ':',
+                style: 'userDataTopic'
+            },
+            {
+                text: user.default_password,
+                style: 'userDataValue'
+            },
+            {
+                text: 'URL:',
+                style: 'userDataTopic'
+            },
+            {
+                text: this.configService.instant<string>('users_pdf_url') || '-',
+                link: this.configService.instant<string>('users_pdf_url'),
+                style: 'userDataValue'
+            },
+            {
+                text: '\n'
+            }
+        ];
+        // url qr code
+        if (this.configService.instant<string>('users_pdf_url')) {
+            columnOpenSlides.push(
+                {
+                    qr: this.configService.instant<string>('users_pdf_url'),
+                    fit: 120,
+                    margin: [0, 0, 0, 8]
+                },
+                {
+                    text: this.translate.instant('Scan this QR code to open URL.'),
+                    style: 'small'
+                }
+            );
+        }
+        return columnOpenSlides;
+    }
+
+    /**
+     * Generates a welcone text according to the events' configuration
+     *
+     * @returns pdfMake definitions
+     */
+    private createWelcomeText(): object {
+        return [
+            {
+                text: this.translate.instant(this.configService.instant<string>('users_pdf_welcometitle')),
+                style: 'userDataHeading'
+            },
+            {
+                text: this.translate.instant(this.configService.instant<string>('users_pdf_welcometext')),
+                style: 'userDataTopic'
+            }
+        ];
+    }
+
+    /**
+     * Handles the creation of the participant lists' table structure
+     *
+     * @param users: passed through to getListUsers
+     * @returns a pdfMake table definition ready to be put into a page
+     */
+    private createUserList(users: ViewUser[]): object {
+        const userTableBody: object[] = [
+            [
+                {
+                    text: '#',
+                    style: 'tableHeader'
+                },
+                {
+                    text: this.translate.instant('Name'),
+                    style: 'tableHeader'
+                },
+                {
+                    text: this.translate.instant('Structure level'),
+                    style: 'tableHeader'
+                },
+                {
+                    text: this.translate.instant('Groups'),
+                    style: 'tableHeader'
+                }
+            ]
+        ];
+        return {
+            table: {
+                widths: ['auto', '*', 'auto', 'auto'],
+                headerRows: 1,
+                body: userTableBody.concat(this.getListUsers(users))
+            },
+            layout: 'headerLineOnly'
```

Comment:
to get a "zebry style" we might want to provide out own layout function.
That's simpler that the attempt I did choose before with the "tableEven" and "tableOdd" function.

``` ts
layout: {
    fillColor: (rowIndex, node, columnIndex) => {
        return (rowIndex % 2 === 0) ? '#EEEEEE' : null;
    }
}
```